frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -647,15 +647,60 @@ func dispatchEnv(d *dispatchState, c *instructions.EnvCommand) error {
 }
 
 func dispatchRun(d *dispatchState, c *instructions.RunCommand, proxy *llb.ProxyEnv, sources []*dispatchState, dopt dispatchOpt) error {
+	var opt []llb.RunOption
+
 	var args []string = c.CmdLine
+	if len(c.Files) > 0 {
+		if len(args) != 1 {
+			panic("Parsing has produced an invalid RunCommand")
```

Comment:
Can we return err rather than panic?

Also, can we include `args` in the err message?

Reply:
Same panic handling in `MustParseHeredoc`

Reply:
It should now return an error here instead of panicking.

R.E. `MustParseHeredoc`, it simply ignores the value of the error - since that's the behavior we kind of expect when calling that variant. Note that the parser handles the errors from badly specified heredocs, so it shouldn't be possible for an error to occur later. Hopefully that's ok, instead of propogating the error back?